R/spatial_block_cv.R

```diff
@@ -98,21 +103,35 @@ spatial_block_cv <- function(data,
     grid_box[3] <- grid_box[3] + abs(grid_box[3] * 0.001)
     grid_box[4] <- grid_box[4] + abs(grid_box[4] * 0.001)
   }
+
+  if (missing(radius)) radius <- NULL
```

Comment:
This is a bit of a kludge: because `radius` and `buffer` get passed through so many functions before getting to `buffer_indices`, I found that they "lost" their missingness at some point along the way. 

I want the user-facing functions to have a default value of 0, but also don't want to spend time on `buffer_indices` unless the user specifically asks for it, so I've been setting these to NULL if missing in the user-facing functions and then to 0 if NULL in `buffer_indices`. If there's a way to just keep these "missing" until they hit `buffer_indices` and then check for missingness instead of NULL there, that would be better.

Reply:
This approach does have the somewhat nice feature that the `radius` and `buffer` attributes are only set when the user specifies them, however.

Reply:
it's relatively easy to lose track of what `radius` and `buffer` are in different stages. Why do you want a user-facing default of `0`?

Reply:
The goal is to emphasize that these functions will not buffer (or add a radius) without user input. This is to highlight the difference from `spatial_buffer_vfold_cv()` (not yet written), which will _require_ at least one of these arguments be non-missing. I could also see setting these to `NULL`, but then I think we'll want to separately document the parameters in each function they're used in

Reply:
Julia cleared up my confusion here; we're going to set these to `NULL` as defaults, and they won't be changed outside of `buffer_indices`